code/datums/actions/build_designator.dm

```diff
@@ -0,0 +1,243 @@
+//List of all images used for constuction designating, in the radial selection menu
+GLOBAL_LIST_INIT(designator_images_list, list(
+	/obj/structure/barricade/solid = image('icons/obj/structures/barricades/metal.dmi', icon_state = "metal_0"),
+	/obj/structure/barricade/solid/plasteel = image('icons/obj/structures/barricades/plasteel.dmi', icon_state = "new_plasteel_0"),
+	/obj/structure/barricade/folding/metal = image('icons/obj/structures/barricades/metal.dmi', icon_state = "folding_metal_0"),
+	/obj/structure/barricade/folding = image('icons/obj/structures/barricades/plasteel.dmi', icon_state = "plasteel_0"),
+))
+
+///Assoc list of construction types to source material
+GLOBAL_LIST_INIT(designator_types, list (
+	/obj/structure/barricade/solid = /obj/item/stack/sheet/metal,
+	/obj/structure/barricade/solid/plasteel = /obj/item/stack/sheet/plasteel,
+	/obj/structure/barricade/folding/metal = /obj/item/stack/sheet/metal,
+	/obj/structure/barricade/folding = /obj/item/stack/sheet/plasteel,
+))
+
+/datum/action/ability/activable/build_designator
+	name = "Construction Designator"
+	desc = "Place a designator for construction."
+	action_icon_state = "build_designator"
+	action_icon = 'icons/mob/actions.dmi'
+	target_flags = ABILITY_TURF_TARGET
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_ABILITY_PLACE_HOLOGRAM,
+		KEYBINDING_ALTERNATE = COMSIG_ABILITY_SELECT_BUILDTYPE,
+	)
+	///personal hologram designator
+	var/obj/effect/build_hologram/hologram
+	/// timerid before we cleanup the hologram
+	var/cleanup_timer
+	/// how long a hologram lasts without movement
+	var/cleanup_time = 4 SECONDS
+	///The typepath of what we want to construct. Typecast for initial var values
+	var/obj/construct_type
+
+/datum/action/ability/activable/build_designator/Destroy()
+	QDEL_NULL(hologram)
+	delete_timer()
+	return ..()
+
+/datum/action/ability/activable/build_designator/should_show()
+	. = ..()
+	if(!.)
+		return
+	return owner.skills.getRating(SKILL_LEADERSHIP) >= SKILL_LEAD_TRAINED
+
+/datum/action/ability/activable/build_designator/can_use_action()
+	return owner.skills.getRating(SKILL_LEADERSHIP) >= SKILL_LEAD_TRAINED
+
+/datum/action/ability/activable/build_designator/fail_activate()
+	if(owner)
+		owner << span_warning("You are not competent enough to do that.") // This message shouldn't show since incompetent people shouldn't have the button, but JIC.
```

Comment:
uhh to chat?

Reply:
do we even need this if it should never happen and does nothing

Reply:
true